woocommerce/woocommerce
# Enforce Strict `@types` Dependencies

### All Submissions:






### Changes proposed in this Pull Request:

As part of my work on #37280, I will be centralizing our TypeScript configuration to make maintenance easier. In the process of doing that, however, I found that there were some `@types` declarations missing from a few packages. By default, TypeScript will search for types in `./node_modules` and will traverse parent directories looking for the same. This can cause issues when a `@types` declaration is missing from the package but a parent directory has one from a different version of the package. 

This pull request isolates the types to the project's directory, and as a consequence, caused a number of type errors across the monorepo. This pull request contains fixes for all of those errors.

In the interesting of an easier time reviewing and testing, this pull request does not contain any TypeScript refactoring changes. The changes in this PR are **just** the result of fixing any errors introduced by setting `"typeRoots"` in all of the `tsconfig.json` and `tsconfig-cjs.json` files.



### How to test the changes in this Pull Request:


* Verify that all of the CI checks pass.
* Verify that the output of `pnpm run test --filter="!woocommerce"` matches what was present before this PR.
* Verify that WooCommerce's React admin pages still work as expected.



### Other information:




### FOR PR REVIEWER ONLY:
